packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -376,8 +387,17 @@ export const useGridVirtualScroller = () => {
       if (panel) {
         rows.push(panel);
       }
+      if (isInfiniteLoadingEnabled && isLastVisible) {
+        rows.push(
+          <InfiniteLoadingTriggerElement
+            ref={(apiRef as any).current.unstable_infiniteLoadingTriggerRef}
+            // Force rerender on last row change to start observing the new trigger
+            key={`trigger-${id}`}
+            role="presentation"
+          />,
+        );
+      }
```

Comment:
I think I meant the `isLastVisible` flag to mean the last logical row in the page, not the last displayed row on screen. Maybe we could move this outside of the loop and just add the element at the end of `rows` when needed?

Reply:
> I think I meant the isLastVisible flag to mean the last logical row in the page, not the last displayed row on screen.

Correct, and for infinite loading, I need to put the trigger element after the last logical row in the page, not the last row rendered due to virtualization.